RHIDP-6499: Modularize Customizing the Learning Paths (#1068)
redhat-developer/red-hat-developers-documentation-rhdh: modules/customizing-the-learning-paths/proc-customizing-the-learning-paths-by-using-hosted-json-files.adoc

[id='proc-customizing-the-learning-paths-by-using-a-hosted-json-file_{context}']
= Customizing the Learning Paths by using a hosted JSON file

For ease of use and simplicity, you can configure the Learning Paths by using a hosted JSON file.

.Procedure

> . Publish the JSON file containing your Learning Paths data to a web server, such as GitHub or Gitlab. You can find an example at link:https://raw.githubusercontent.com/redhat-developer/rhdh/main/packages/app/public/learning-paths/data.json[].

Author: ```suggestion
. Publish the JSON file containing your Learning Paths data to a web server, such as GitHub or Gitlab. You can find an example at link:https://raw.githubusercontent.com/redhat-developer/rhdh/release-{product-version}/packages/app/public/learning-paths/data.json[].
```

> . Configure the {product-short} proxy to access the Learning Paths data from the hosted JSON file, by adding the following code to the `{my-app-config-file}` file:

Author: ```suggestion
. Configure the {product-short} proxy to access the Learning Paths data from the hosted JSON file, by adding the following to the `{my-app-config-file}` file:
```

+
[source,yaml]
----
proxy:
  endpoints:
    '/developer-hub':
      target: https://raw.githubusercontent.com/
      pathRewrite:
        '^/api/proxy/developer-hub/learning-paths': '/redhat-developer/rhdh/main/packages/app/public/learning-paths/data.json'
      changeOrigin: true
      secure: true

> ----

Author: ```suggestion
. Publish the JSON file containing your Learning Paths data to a web server, such as GitHub or Gitlab. You can find an example at link:https://raw.githubusercontent.com/redhat-developer/rhdh/release-{product-version}/packages/app/public/learning-paths/data.json[].

. Configure the {product-short} proxy to access the Learning Paths data from the hosted JSON file, by adding the following to the `{my-app-config-file}` file:
+
[source,yaml,subs='+quotes']
----
proxy:
  endpoints:
    '/developer-hub':
      target: _<target>_
      pathRewrite:
        '^/api/proxy/developer-hub/learning-paths': '_<learning_path.json>_'
      changeOrigin: true
      secure: true
----

`_<target>_`:: Enter the hosted JSON file base URL, such as `https://raw.githubusercontent.com`.

`_<learning_path.json>_`:: Enter the hosted JSON file path without the base URL, such as
`'/redhat-developer/rhdh/main/packages/app/public/learning-paths/data.json'`
```

> [id='proc-customize-rhdh-learning-paths_{context}']
> = Customizing the Learning Paths by using hosted JSON files
> 
> For ease of use and simplicity, you can configure the Learning Paths by using hosted JSON files.
> 
> 
> .Procedure
> . Publish the JSON files containing your Learning Paths data to a web server, such as GitHub or Gitlab. You can find an example at link:https://raw.githubusercontent.com/redhat-developer/rhdh/main/packages/app/public/learning-paths/data.json[].
> 
> . Configure the {product-short} proxy to access the Learning Paths data from the hosted JSON files, by adding the following code to the `{my-app-config-file}` file:

Author: ```suggestion
[id='proc-customizing-the-learning-paths-by-using-a-hosted-json-file_{context}']
= Customizing the Learning Paths by using a hosted JSON file

For ease of use and simplicity, you can configure the Learning Paths by using a hosted JSON file.

.Procedure
. Publish the JSON file containing your Learning Paths data to a web server, such as GitHub or Gitlab. You can find an example at link:https://raw.githubusercontent.com/redhat-developer/rhdh/main/packages/app/public/learning-paths/data.json[].

. Configure the {product-short} proxy to access the Learning Paths data from the hosted JSON file, by adding the following code to the `{my-app-config-file}` file:
```
